PR: https://github.com/odoo/odoo/pull/225077

From: 94c8debc5f6473a3fdb8b9db4de7ef8674129f99
From: std-odoo
Date: 2026-01-09 13:40:15

Breaking data model changes scores: del:6 + add:0, change matches:
-    _inherit = 'res.partner'
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string',
-    iap_search_domain = fields.Char('Search Domain / Email',
-    partner_id = fields.Many2one('res.partner', string='Partner',
-    iap_search_domain = fields.Char('Search Domain / Email', help='Domain used to find the company')
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string', readonly=True)

Total Changes: 100

[IMP] (*_)mail_plugin: remove the enrichment feature from the addin

Purpose
=======
The enrichment part of the addon was slow, and made the plugin
annoying to use. We want now to focus on the logging part of the
plugin, and to simplify everything in that direction.

Now that we removed the enrichment part, a lot of things can be
simplified on the Odoo side.

Task-4727609

Part-of: odoo/odoo#225077
Related: odoo/enterprise#93676
Related: odoo/upgrade#8369
Signed-off-by: Thibault Delavallee (tde) <tde@openerp.com>

================================= pseudo patch: =================================

--- a/addons/mail_plugin/models/__init__.py
+++ b/addons/mail_plugin/models/__init__.py
@@ -2,5 +2,3 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
 from . import ir_http
-from . import res_partner
-from . import res_partner_iap

--- a/addons/mail_plugin/models/res_partner.py
+++ b/None
@@ -1,68 +0,0 @@
-# -*- coding: utf-8 -*-
-from odoo import api, fields, models
-
-
-class ResPartner(models.Model):
-    _inherit = 'res.partner'
-
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string',
-                                  compute='_compute_partner_iap_info')
-
-    iap_search_domain = fields.Char('Search Domain / Email',
-                                compute='_compute_partner_iap_info')
-
-    def _compute_partner_iap_info(self):
-        partner_iaps = self.env['res.partner.iap'].sudo().search([('partner_id', 'in', self.ids)])
-        partner_iaps_per_partner = {
-            partner_iap.partner_id: partner_iap
-            for partner_iap in partner_iaps
-        }
-
-        for partner in self:
-            partner_iap = partner_iaps_per_partner.get(partner)
-            if partner_iap:
-                partner.iap_enrich_info = partner_iap.iap_enrich_info
-                partner.iap_search_domain = partner_iap.iap_search_domain
-            else:
-                partner.iap_enrich_info = False
-                partner.iap_search_domain = False
-
-    @api.model_create_multi
-    def create(self, vals_list):
-        partners = super().create(vals_list)
-        # Not done with inverse method so we do not need to search
-        # for existing <res.partner.iap>
-        partner_iap_vals_list = [{
-            'partner_id': partner.id,
-            'iap_enrich_info': vals.get('iap_enrich_info'),
-            'iap_search_domain': vals.get('iap_search_domain'),
-        } for partner, vals in zip(partners, vals_list) if vals.get('iap_enrich_info') or vals.get('iap_search_domain')]
-        self.env['res.partner.iap'].sudo().create(partner_iap_vals_list)
-        return partners
-
-    def write(self, vals):
-        res = super(ResPartner, self).write(vals)
-
-        if 'iap_enrich_info' in vals or 'iap_search_domain' in vals:
-            # Not done with inverse method so we do need to search
-            # for existing <res.partner.iap> only once
-            partner_iaps = self.env['res.partner.iap'].sudo().search([('partner_id', 'in', self.ids)])
-            missing_partners = self
-            for partner_iap in partner_iaps:
-                if 'iap_enrich_info' in vals:
-                    partner_iap.iap_enrich_info = vals['iap_enrich_info']
-                if 'iap_search_domain' in vals:
-                    partner_iap.iap_search_domain = vals['iap_search_domain']
-
-                missing_partners -= partner_iap.partner_id
-
-            if missing_partners:
-                # Create new <res.partner.iap> for missing records
-                self.env['res.partner.iap'].sudo().create([
-                    {
-                        'partner_id': partner.id,
-                        'iap_enrich_info': vals.get('iap_enrich_info'),
-                        'iap_search_domain': vals.get('iap_search_domain'),
-                    } for partner in missing_partners
-                ])
-        return res

--- a/addons/mail_plugin/models/res_partner_iap.py
+++ b/None
@@ -1,30 +0,0 @@
-# -*- coding: utf-8 -*-
-
-
-from odoo import fields, models
-
-
-class ResPartnerIap(models.Model):
-    """Technical model which stores the response returned by IAP.
-
-    The goal of this model is to not enrich 2 times the same company. We do it in a
-    separate model to not add heavy field (iap_enrich_info) on the <res.partner>
-    model.
-
-    We also save the requested domain, so whatever the values are on the <res.partner>,
-    we will always retrieve the already enriched <res.partner> and the corresponding
-    IAP information.
-    """
-
-    _name = 'res.partner.iap'
-    _description = 'Partner IAP'
-
-    partner_id = fields.Many2one('res.partner', string='Partner',
-                                 ondelete='cascade', required=True)
-    iap_search_domain = fields.Char('Search Domain / Email', help='Domain used to find the company')
-    iap_enrich_info = fields.Text('IAP Enrich Info', help='IAP response stored as a JSON string', readonly=True)
-
-    _unique_partner_id = models.Constraint(
-        'UNIQUE(partner_id)',
-        'Only one partner IAP is allowed for one partner',
-    )
